### PR TITLE
Updates to support xcode 16

### PR DESCRIPTION
**Pull request by ztefanie**
### Short description
Adujsted project so it builds on Xcode 16

### Proposed changes
Updated sentry to fix this: https://github.com/getsentry/sentry-react-native/issues/3883
Updated url launcher to fix this https://github.com/flutter/flutter/issues/152986

### Side effects
- May break anything related to any library

### Testing
smoke test the whole app on iOS and android
check if the app builds on Xcode 16 and 15 for you please

### Resolved issues
no issue, but will me merged into / is needed for #1431

### CODE REVIEW

**Comment by f1sh1918** [comment]
@ztefanie 
i made some smoke tests and everything except of deeplinking activation works.
Maybe you can merge the recent changes from main (where its working) your working branch

The issue i'm facing here is that the link doesn't work with trailing slash anymore (even on android) and on main it does
```js
npx uri-scheme open berechtigungskarte://bayern.ehrenamtskarte.app/activation/code#CkkKHwoPQW5kcmVhcyBGaXNjaGVyEPGkARoICgIICSICCAASEDGjkYQBPaU2Dd6mI87eXtAaFHNsrKCgpGu20z2c06yyG7bU12ap/ --ios
```

**Review by f1sh1918 — APPROVED** [review]
Smoke tested app tabs and activation via deeplinking

**Comment by ztefanie** [comment]
I also tested deeplinking, works as expected. Will merge this

File: frontend/pubspec.lock [hunk 1]
```diff
@@ -349,50 +349,50 @@ packages:
     dependency: "direct main"
     description:
       name: flutter_secure_storage
-      sha256: ffdbb60130e4665d2af814a0267c481bcf522c41ae2e43caf69fa0146876d685
+      sha256: "165164745e6afb5c0e3e3fcc72a012fb9e58496fb26ffb92cf22e16a821e85d0"
       url: "https://pub.dev"
     source: hosted
-    version: "9.0.0"
```
[the remark this answers is not in the document]
Comment (michael-markl) [reply frontend/pubspec.lock:355]: See https://github.com/mogol/flutter_secure_storage/issues/762

[the remark this answers is not in the document]
Comment (michael-markl) [reply frontend/pubspec.lock:355]: And https://github.com/digitalfabrik/entitlementcard/issues/1265

**Pull request merged**